scripts/lint-frontend-messages.sh

```
@@ -0,0 +1,37 @@
+#
+# ESLint の no-restricted-syntax は throw new Error の AST しか拾えないため、
+# 関数呼び出し系はこのスクリプトで補完する。
+#
+# 詳細: .claude/rules/frontend/messages.md
+set -euo pipefail
+
+cd "$(dirname "$0")/../frontend"
+
+# Rust regex (ripgrep -P) で Unicode プロパティを使って ひらがな・カタカナ・漢字 を検出する。
+# シングルクォート文字列とバッククォート文字列の両方をカバー。
+PATTERN='(set\w*Error\w*|toast\.error|alert)\(\s*["`][^"`]*[\p{Hiragana}\p{Katakana}\p{Han}]'
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)